### DISCUSSION TITLE
Separating business shared folders from main onedrive content

### DISCUSSION THREAD

**Discussion by digehode** [abraunegg/onedrive#1257] [Q&A] [answered]
I can successfully sync my own files using the tool and all is well. I can also get it to sync the business shared folders. 

My first attempt unfortunately led to the business shared folders being placed in the same local directory as my own files, so at next sync they were being copied to the onedrive service as new files there. This obviously isn't what I want.

I've created a new config dir and authorised the client, but I can't work out how to turn **off** the syncing of everything **except** the business shared folders.  In a dry run, it always shows that it will sync my own files and directories too, leading to the same issue.

I think I could do this with selective sync on the command-line but I have dozens of these business folders and ideally just want a config that can be easily daemonised to deal with the whole category.

I expected this to be detailed in the docs at https://github.com/abraunegg/onedrive/blob/master/docs/advanced-usage.md but couldn't see it.  Everything is so nicely documented and easy to follow, I can't help but feel I might just be missing where this is recorded.

I hope this is the right place to ask about this.

**Comment by abraunegg** [comment, marked as answer]
@digehode 
> I've created a new config dir and authorised the client, but I can't work out how to turn off the syncing of everything except the business shared folders. In a dry run, it always shows that it will sync my own files and directories too, leading to the same issue.

The only way at present is to have 2 separate configurations:

**Client Configuration 1: Normal OneDrive account folders only**
* Use 'sync_list' to only sync the folders in your OneDrive account root
* Configure `sync_business_shared_folders = "false"` within your config file
     
**Client Configuration 2: Sync Business Folders only**
* Configure `sync_business_shared_folders = "true"` within your config file
* Use 'sync_list' to only sync the Business Shared Folder names

Test this with `--dry-run` to validate this will work as expected before using without.

Feature Request https://github.com/abraunegg/onedrive/issues/1251 will (when I get to it) may simplify this, but at present the above is your only option.

**Reply by digehode** [reply]
Ah, yes, I see. I didn't think of this. I was still thinking of them being separate but this makes sense if the tool combined the normal//business folder paths.

For anyone else reading this, it was also possible to symlink the business_shared_folders file to sync_list so that I only had one list of what I wanted.

@abraunegg: thanks!

**Reply by abraunegg** [reply]
@digehode 
> For anyone else reading this, it was also possible to symlink the business_shared_folders file to sync_list so that I only had one list of what I wanted.

The only issue with this approach, is the following:
* 'business_shared_folders' file, the line entries are checked / searched at the start of the path for an inclusion match
* 'sync_list' file, the line entries are checked / searched for any inclusion match

Consider the following file|folder paths:

* `/Documents/next/level/folder/` <- Business Shared Folder
* `/some_folder/next/level/folder/Documents/` <- Normal non-shared folder

If the entries in both 'business_shared_folders' & 'sync_list' is simply:
```
Documents
```

There is the potential for *both* paths to match syncing (I have not tested to confirm .. but I strongly suspect this is the case)

To resolve this, 'sync_list' would need the following tweak:
```
Documents/*
```

Thus, symbolic linking the two files is not advisable.
